extensions/azurecore/src/account-provider/auths/azureAuth.ts

```diff
@@ -662,17 +812,41 @@ export abstract class AzureAuth implements vscode.Disposable {
 
 	public async clearCredentials(account: azdata.AccountKey): Promise<void> {
 		try {
-			return this.deleteAccountCache(account);
+			// remove account based on authLibrary field, accounts added before this field was present will default to
+			// ADAL method of account removal
+			if (account.authLibrary === 'MSAL') {
```

Comment:
there is inconsistent "default" case in the PR, where typically MSAL is the else block and here ADAL is.  This could be an issue if the setting was some random value (not MSAL or ADAL).